docs/src/man/comparisons.md

```diff
@@ -235,6 +235,59 @@ The table below compares more advanced commands:
 | DataFrame as input        | `summarize(df, head(across(), 2))`                        | `combine(d -> first(d, 2), df)`                               |
 | DataFrame as output       | `summarize(df, tibble(value = c(min(x), max(x))))`        | `combine(:x => x -> (value = [minimum(x), maximum(x)],), df)` |
 
+## Comparison with the R package data.table
+
+The following table compares the main functions of DataFrames.jl with the R package data.table (version 1.14.1). To contrast data.table with dplyr, the focus has been put on transformations in place.
+
+|Operations              | data.table  | DataFrames.jl|  
+|:-----------------------|:------------|:-------------|
+|Reduce multiple values  |`df[, list(mean(x))]`       |`combine(df, :x => mean)`|
+|Add new columns         |`df[, x_mean := mean(x) ]`  |`transform(df, :x => mean => :x_mean)`|
+|Rename column (in place) |`setnames(df, "x", "x_new")`|`rename!(df, :x => :x_new)`|
+|Rename multiple columns (in place) |`setnames(df, c("x", "y"), c("x_new", "y_new")`|`rename!(df, [:x, :y] => [:x_new, :y_new])`|
+|Pick columns            |`df[, list(x, y)]`          |`select(df, :x, :y)`|
+|Remove columns          |`df[, -"x" ]`               |`select(df, Not(:x))`|
+|Remove columns (in place) |`df[, c("x") := NULL ]`   |`select!(df, Not(:x))`|
+|Pick & transform columns|`df[, list(mean(x), y)]`    |`select(df, :x => mean, :y)`|
+|Pick rows               |`df[ x >= 1 ]`              |`filter(:x => >=(1), df)`|
+|Sort rows (in place)    |`setorder(df, x)`           |`sort!(df, :x)`|
+|Sort rows               |`df[ order(x) ]`            |`sort(df, :x)`|
+
+### Grouping data and aggregation
+
+|Operations              | data.table                                      | DataFrames.jl     |
+|:-----------------------|:------------------------------------------------|:------------------|
+|Reduce multiple values  |`df[, list(mean(x)), by = list(id) ]`            |`combine(groupby(df, :id), :x => mean)`|
+|Add new columns (in place) |`df[, x_mean := mean(x), by = list(id) ]`        |`transform!(groupby(df, :id), :x => mean)`|
+|Pick & transform columns|`df[, list(x_mean = mean(x), y), by = list(id) ]`|`select(groupby(df, :id), :x => mean, :y)`|
+
+### More advanced commands
+
+|Operations              | data.table                                      | DataFrames.jl             |
+|:-----------------------|:------------------------------------------------|:--------------------------|
+|Complex Function          |`df[, list(mean(x, na.rm = T)) ]`              |`combine(df, :x => x -> mean(skipmissing(x)))`|
+|Transform certain rows (in place) |`df[x<=0, x:=0 ]`                      |`transform!(df, :x => (x -> ifelse.(x .<= 0, 0, x)) => :x)`|
```

Comment:
why not just:
```
df.x[df.x .<= 0] .= 0
```
?

Reply:
I copied the stata example. I have asked this question previously and the ifelse was the one recommended. I think we ought to include one such example where x might include missing (in which case we have to add a coalesce?). 

Tbh, this is where the base data frames syntax shows its limit and I hope that this sort of complicated things will be easier with dataframesmeta down the road.

Reply:
I agree that currently we do not have a super nice syntax for working on subsets of rows. I guess the difference between `df.x[df.x .<= 0] .= 0` and `transform!(df, :x => (x -> ifelse.(x .<= 0, 0, x)) => :x)` is that:

```
df.x[df.x .<= 0] .= 0
```
does not play way with piping, but as `data.table` is not piping oriented either then I thought it is not a problem. I assume @pdeffebach prefers `transform!` because of piping.